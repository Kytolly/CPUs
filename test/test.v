module decoder3_8(
    input wire [2:0] in,
    output reg [7:0] out
);
    // always/initial 模块中只能用 reg型变量
    always @(*) begin
        case(in)
            3'h0: out = 8'h01;
            3'h1: out = 8'h02;
            3'h2: out = 8'h04;
            3'h3: out = 8'h08;
            3'h4: out = 8'h10;
            3'h5: out = 8'h20;
            3'h6: out = 8'h40;
            3'h7: out = 8'h80;
            // 避免lanth
            default: out = 8'h00;
        endcase
    end
endmodule
