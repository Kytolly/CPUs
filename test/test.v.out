#! /usr/bin/vvp
:ivl_version "11.0 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/system.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/va_math.vpi";
S_0x55807077ae20 .scope module, "decoder3_8" "decoder3_8" 2 1;
 .timescale 0 0;
    .port_info 0 /INPUT 3 "in";
    .port_info 1 /OUTPUT 8 "out";
o0x7f48d0f7f018 .functor BUFZ 3, C4<zzz>; HiZ drive
v0x55807077cba0_0 .net "in", 2 0, o0x7f48d0f7f018;  0 drivers
v0x5580707c3530_0 .var "out", 7 0;
E_0x55807077b420 .event edge, v0x55807077cba0_0;
    .scope S_0x55807077ae20;
T_0 ;
    %wait E_0x55807077b420;
    %load/vec4 v0x55807077cba0_0;
    %dup/vec4;
    %pushi/vec4 0, 0, 3;
    %cmp/u;
    %jmp/1 T_0.0, 6;
    %dup/vec4;
    %pushi/vec4 1, 0, 3;
    %cmp/u;
    %jmp/1 T_0.1, 6;
    %dup/vec4;
    %pushi/vec4 2, 0, 3;
    %cmp/u;
    %jmp/1 T_0.2, 6;
    %dup/vec4;
    %pushi/vec4 3, 0, 3;
    %cmp/u;
    %jmp/1 T_0.3, 6;
    %dup/vec4;
    %pushi/vec4 4, 0, 3;
    %cmp/u;
    %jmp/1 T_0.4, 6;
    %dup/vec4;
    %pushi/vec4 5, 0, 3;
    %cmp/u;
    %jmp/1 T_0.5, 6;
    %dup/vec4;
    %pushi/vec4 6, 0, 3;
    %cmp/u;
    %jmp/1 T_0.6, 6;
    %dup/vec4;
    %pushi/vec4 7, 0, 3;
    %cmp/u;
    %jmp/1 T_0.7, 6;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v0x5580707c3530_0, 0, 8;
    %jmp T_0.9;
T_0.0 ;
    %pushi/vec4 1, 0, 8;
    %store/vec4 v0x5580707c3530_0, 0, 8;
    %jmp T_0.9;
T_0.1 ;
    %pushi/vec4 2, 0, 8;
    %store/vec4 v0x5580707c3530_0, 0, 8;
    %jmp T_0.9;
T_0.2 ;
    %pushi/vec4 4, 0, 8;
    %store/vec4 v0x5580707c3530_0, 0, 8;
    %jmp T_0.9;
T_0.3 ;
    %pushi/vec4 8, 0, 8;
    %store/vec4 v0x5580707c3530_0, 0, 8;
    %jmp T_0.9;
T_0.4 ;
    %pushi/vec4 16, 0, 8;
    %store/vec4 v0x5580707c3530_0, 0, 8;
    %jmp T_0.9;
T_0.5 ;
    %pushi/vec4 32, 0, 8;
    %store/vec4 v0x5580707c3530_0, 0, 8;
    %jmp T_0.9;
T_0.6 ;
    %pushi/vec4 64, 0, 8;
    %store/vec4 v0x5580707c3530_0, 0, 8;
    %jmp T_0.9;
T_0.7 ;
    %pushi/vec4 128, 0, 8;
    %store/vec4 v0x5580707c3530_0, 0, 8;
    %jmp T_0.9;
T_0.9 ;
    %pop/vec4 1;
    %jmp T_0;
    .thread T_0, $push;
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "test.v";
